### PR TITLE
Cleanup GTM dependency from Firebase performance since we don't need log filters anymore

### LINKED ISSUE #6597: GoogleUtilitiesLegacy conflicting methods in categories
### Step 1: Describe your environment

  * Xcode version: `12.0`
  * Firebase SDK version: `6.33.0`
  * Firebase Component: `Performance`
  * Component version: `6.33.0`
  * Installation method: `CocoaPods`

### Step 2: Describe the problem

When `Firebase/Performance` is used with `GoogleTagManager`, Xcode 12 generates warnings about conflicting methods in different categories:

```
ld: warning: method '-logMessage:level:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMLogger.o) conflicts with same method from another category
ld: warning: method '+fileHandleForLoggingAtPath:mode:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMLogger.o) conflicts with same method from another category
ld: warning: method '-logMessage:level:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMLogger.o) conflicts with same method from another category
ld: warning: method '+gtm_dataByGzippingBytes:length:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMNSData+zlib.o) conflicts with same method from another category
ld: warning: method '+gtm_dataByGzippingData:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMNSData+zlib.o) conflicts with same method from another category
ld: warning: method '+gtm_dataByGzippingBytes:length:compressionLevel:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMNSData+zlib.o) conflicts with same method from another category
ld: warning: method '+gtm_dataByGzippingData:compressionLevel:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMNSData+zlib.o) conflicts with same method from another category
ld: warning: method '+gtm_dataByDeflatingBytes:length:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMNSData+zlib.o) conflicts with same method from another category
ld: warning: method '+gtm_dataByDeflatingData:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMNSData+zlib.o) conflicts with same method from another category
ld: warning: method '+gtm_dataByDeflatingBytes:length:compressionLevel:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMNSData+zlib.o) conflicts with same method from another category
ld: warning: method '+gtm_dataByDeflatingData:compressionLevel:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMNSData+zlib.o) conflicts with same method from another category
ld: warning: method '+gtm_dataByInflatingBytes:length:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMNSData+zlib.o) conflicts with same method from another category
ld: warning: method '+gtm_dataByInflatingData:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMNSData+zlib.o) conflicts with same method from another category
ld: warning: method '+gtm_dataByRawDeflatingBytes:length:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMNSData+zlib.o) conflicts with same method from another category
ld: warning: method '+gtm_dataByRawDeflatingData:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMNSData+zlib.o) conflicts with same method from another category
ld: warning: method '+gtm_dataByRawDeflatingBytes:length:compressionLevel:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMNSData+zlib.o) conflicts with same method from another category
ld: warning: method '+gtm_dataByRawDeflatingData:compressionLevel:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMNSData+zlib.o) conflicts with same method from another category
ld: warning: method '+gtm_dataByRawInflatingBytes:length:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMNSData+zlib.o) conflicts with same method from another category
ld: warning: method '+gtm_dataByRawInflatingData:' in category from /Users/user/develop/_bugreport/google_utils_conflict/Pods/GoogleUtilitiesLegacy/Frameworks/frameworks/GoogleUtilitiesLegacy.framework/GoogleUtilitiesLegacy(GTMNSData+zlib.o) conflicts with same method from another category
```

#### Steps to reproduce:

Add to Podfile:
```ruby
    pod 'GoogleTagManager'
    pod 'Firebase/Performance'
```

Example project:
https://github.com/voidless/google_utils_conflict

### PR DESCRIPTION
**Pull request by visumickey**
Fix #6597

### CODE REVIEW

File: FirebasePerformance.podspec [hunk 1]
```diff
@@ -60,7 +60,6 @@ Firebase Performance library to measure performance of Mobile and Web Apps.
   s.dependency 'FirebaseInstallations', '~> 7.0'
   s.dependency 'FirebaseRemoteConfig', '~> 7.0'
   s.dependency 'GoogleDataTransport', '~> 8.2'
-  s.dependency 'GTMSessionFetcher/Core', '~> 1.1'
   s.dependency 'GoogleToolboxForMac/Logger', '~> 2.1'
```
Comment (paulb777) [inline FirebasePerformance.podspec:63]: The `GoogleToolboxForMac` is the actual problem in #6597.  

GTMSessionFetcher is used by a few other Firebase libraries and doesn't cause similar problems.  Still great to remove to slim things down.

**Review by paulb777 — COMMENTED** [review]
Please add a Changelog entry about the GTMSessionFetcher dependency removal. Dependency removal can cause problems especially for users using binary distributions and not doing a clean reinstall.

File: FirebasePerformance.podspec (hunk 1 above)
Comment (visumickey) [reply FirebasePerformance.podspec]: Good catch! Cleaned up the dependency on GoogleToolboxForMac as well. Apparently, we are using the FIRLogging service and nothing specifically from GTM library.

File: FirebasePerformance/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# Version 7.4.0
+* Make FirebasePerformance open source.
+* Remove GoogleToolboxForMac dependency.
```
Comment (paulb777) [inline FirebasePerformance/CHANGELOG.md:3]: and GTMSesssionFetcher

**Review by paulb777 — APPROVED** [review]

File: FirebasePerformance/CHANGELOG.md (hunk 2 above)
Comment (visumickey) [reply FirebasePerformance/CHANGELOG.md]: > and GTMSesssionFetcher

Makes sense. Done!

**Review by jeremyjiang-dev — APPROVED** [review]

**Review by zijianjoy — APPROVED** [review]

**Pull request merged**